test/rekt/features/apiserversource/data_plane.go

```diff
@@ -766,3 +767,58 @@ func SendsEventsWithRetries() *feature.Feature {
 			).AtLeast(1))
 	return f
 }
+
+func SendsEventsWithBrokerAsSinkTLS() *feature.Feature {
+	src := feature.MakeRandomK8sName("apiserversource")
+	sacmName := feature.MakeRandomK8sName("apiserversource")
+	brokerName := feature.MakeRandomK8sName("broker")
+	sinkName := feature.MakeRandomK8sName("sink")
+	triggerName := feature.MakeRandomK8sName("trigger")
+	f := feature.NewFeature()
+
+	f.Prerequisite("transport encryption is strict", featureflags.TransportEncryptionStrict())
+	f.Prerequisite("should not run when Istio is enabled", featureflags.IstioDisabled())
+
+	f.Setup("install broker", broker.Install(brokerName, broker.WithEnvConfig()...))
+	f.Setup("broker is ready", broker.IsReady(brokerName))
+	f.Setup("broker is addressable", broker.IsAddressable(brokerName))
+	f.Setup("Broker has HTTPS address", broker.ValidateAddress(brokerName, addressable.AssertHTTPSAddress))
+
+	f.Setup("install sink", eventshub.Install(sinkName, eventshub.StartReceiverTLS))
+
+	f.Setup("install trigger", func(ctx context.Context, t feature.T) {
+		d := service.AsDestinationRef(sinkName)
+		d.CACerts = eventshub.GetCaCerts(ctx)
+		trigger.Install(triggerName, brokerName, trigger.WithSubscriberFromDestination(d))(ctx, t)
+	})
+	f.Setup("Wait for Trigger to become ready", trigger.IsReady(triggerName))
+
+	f.Setup("Create Service Account for ApiServerSource with RBAC for v1.Event resources",
+		setupAccountAndRoleForPods(sacmName))
+
+	cfg := []manifest.CfgFn{
+		apiserversource.WithServiceAccountName(sacmName),
+		apiserversource.WithEventMode(v1.ResourceMode),
+		apiserversource.WithResources(v1.APIVersionKindSelector{
+			APIVersion: "v1",
+			Kind:       "Event",
+		}),
+	}
+
+	f.Requirement("install ApiServerSource", func(ctx context.Context, t feature.T) {
+		d := broker.AsDestinationRef(brokerName)
+		d.CACerts = eventshub.GetCaCerts(ctx)
```

Comment:
Actually you have to use the brokers CACerts to be correct here. I guess you can use the `broker.Address(ctx, brokerName)` function for this.

Reply:
If I understand Pierangelo's comment correctly [here](https://github.com/knative/eventing/pull/7096#discussion_r1272520044), I used broker.Address orginally. @creydr

Reply:
I didn't mean to use `d.URI`. The `broker.AsDestinationRef()` way is fine. I mean, that the brokers CACerts can be different from the eventshubs CACerts IIUC.
I mean something like:

```
d := broker.AsDestinationRef(brokerName)
brokerAddr, err := broker.Address(ctx, brokerName)
d.CACerts = brokerAddr.CACerts
...
```